sdk/core/azure-core/azure/core/tracing/common.py

```diff
@@ -0,0 +1,91 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+from os import environ
+import re
+
+from azure.core.tracing.context import tracing_context
+from azure.core.tracing.abstract_span import AbstractSpan
+from azure.core.settings import settings
+
+
+def _get_opencensus_wrapper():
+    # type: () -> OpencensusWrapper
+    """Returns the OpencensusWrapper if opencensus is installed else returns None"""
+    try:
+        from azure.core.tracing.ext.opencensus_wrapper import OpencensusWrapper
+
+        return OpencensusWrapper
+    except ImportError:
+        return None
+
+
+def set_span_contexts(wrapped_span, span_instance=None, impl_wrapper=None):
+    # type: (AbstractSpan, AbstractSpan) -> None
+    tracing_context.current_span.set(wrapped_span)
+    impl_wrapper = impl_wrapper or wrapped_span
+    tracing_context.tracing_impl.set(impl_wrapper.__class__)
+    if wrapped_span is not None or (
+        span_instance is not None and impl_wrapper is not None
+    ):
+        span_instance = span_instance or wrapped_span.span_instance
+        impl_wrapper.set_current_span(span_instance)
```

Comment:
This conditional is hard for me to follow. I think this is equivalent (set from `wrapped_span` when there is one?)
```
if wrapped_span:
    span_instance = wrapped_span.span_instance

if impl_wrapper and span_instance:
    impl_wrapper.set_current_span(span_instance)
```

Reply:
actually I guess the original will keep `span_instance` over `wrapped_span.span_instance`, in which case:
```
if wrapped_span and not span_instance:
    span_instance = wrapped_span.span_instance

```

Reply:
ok you are right. I think I fixed it. Let me know.